src/Processors/Processor.cs

```diff
@@ -3,19 +3,52 @@
 public abstract class Processor : BaseThread
 {
     private bool _isPaused = false;
+    private bool _onPauseOnce = false;
+    private bool _onResumeOnce = false;
     private bool _stopRequested = false;
     private int _interval = 500;
 
+    protected Processor()
+    {
+        OnCreate();
+    }
+
+    public abstract void OnCreate();

```

Comment:
Could you make the override of all these abstract methods optional? except for Process

Reply:
Done! Thank you for the advice.